src/test/dependencies.test.ts

```diff
@@ -17,5 +35,40 @@ suite("dependencies", () => {
         // just demonstrating that it doesn't crash; this is the example from the readme.
     });
 
-    // TODO test the actual installer components.
+    test("HTTP Download", function() {
+        this.timeout(HTTP_DOWNLOAD_TIMEOUT_MS);
+        const myDependency = new Dependency<"remote">(
+            TMP_PATH,
+            ["remote",
+              new InstallerSequence([
+                    new FileDownloader(HTTP_DOWNLOAD_URL)
+              ])
+            ]
+          );
+        return withProgressInWindow(
+            "Testing download of an HTTP file",
+            listener => myDependency.install("remote", true, listener));
+    });
+
+    test("HTTPS Download", function() {
+        this.timeout(HTTPS_DOWNLOAD_TIMEOUT_MS);
+        const myDependency = new Dependency<"remote">(
+            TMP_PATH,
+            ["remote",
+              new InstallerSequence([
+                    new FileDownloader(HTTPS_DOWNLOAD_URL)
+              ])
+            ]
+          );
```

Comment:
The indentation here feels a little inconsistent: 4 spaces everywhere but 2 in the dependency definitions. Looks like that would end up changing as soon as someone auto-formatted the file.